test/command/parser/parser_test.exs

```diff
@@ -298,6 +300,43 @@ defmodule Parser.ParserTest do
     should_not_parse "foo --opt1=$blah[3.wubba"
   end
 
+  test "unicode args aren't mangled" do
+    {:ok, ast} = Parser.scan_and_parse("echo #{@unicode_text}")
+    args = ast.stages.left.args
+    assert Enum.count(args) == 1
+    first_arg = Enum.at(args, 0)
+    assert first_arg.__struct__ == Piper.Command.Ast.String
+    assert first_arg.value == "#{@unicode_text}"
+    assert "#{ast}" == "echo #{@unicode_text}"
+  end
+
+  test "mixed ascii and unicode args aren't mangled" do
+    {:ok, ast} = Parser.scan_and_parse("echo #{@unicode_text} there!")
+    assert "#{ast}" == "echo #{@unicode_text} there!"
+    {:ok, ast} = Parser.scan_and_parse("echo To you I say #{@unicode_text} 1 time")
+    args = ast.stages.left.args
+    assert Enum.count(args) == 7
+    assert "#{ast}" == "echo To you I say #{@unicode_text} 1 time"
+  end
+
+  test "unicode option values aren't mangled" do
+    {:ok, ast} = Parser.scan_and_parse("echo --message=\"#{@unicode_text}\"")
+    invocation = ast.stages.left
+    assert Enum.count(invocation.args) == 1
+    opt = Enum.at(invocation.args, 0)
+    assert opt.__struct__ == Piper.Command.Ast.Option
+    assert opt.value.value == "#{@unicode_text}"
+    assert "#{ast}" == "echo --message=\"#{@unicode_text}\""
+  end
+
+  test "unicode command names aren't mangled" do
+    {:error, "illegal characters \"\u1E23\""} = Parser.scan_and_parse("#{@unicode_text} 123")
```

Comment:
Does this mean we can't have unicode in command names? We might need to add a validation check for this when installing a bundle.